pkg/controller/kogitoapp/resource/manage_resources.go

```diff
@@ -0,0 +1,360 @@
+
+		if updateS2I {
+			log.Info("Updating BuildConfig for S2I")
+
+			if err := kubernetes.ResourceC(client).Update(buildConfigS2I); err != nil {
+				return updateS2I, err
+			}
+
+			if _, err := openshift.BuildConfigC(client).TriggerBuild(buildConfigS2I, instance.Name); err != nil {
+				return updateS2I, err
+			}
+
+			return updateS2I, nil
+		}
+
+	}
+
+	return false, nil
+}
+
+func ensureBuildConfigRuntime(instance *v1alpha1.KogitoApp, buildConfigRuntime *buildv1.BuildConfig, client *client.Client) (bool, error) {
+	if buildConfigRuntime != nil {
+		log := log.With("kind", buildConfigRuntime.GetObjectKind().GroupVersionKind().Kind, "name", buildConfigRuntime.Name, "namespace", buildConfigRuntime.Namespace)
+
+		if _, err := kubernetes.ResourceC(client).Fetch(buildConfigRuntime); err != nil {
+			return false, err
+		}
+
+		updateRuntime := false
+
+		runtimeImage, buildType := resolveRuntimeImage(instance)
+		if buildConfigRuntime.Labels[LabelKeyBuildType] != string(buildType) {
+			log.Debug("Changes detected in 'BuildType'.", " OLD - ", buildConfigRuntime.Labels[LabelKeyBuildType], " NEW - ", buildType)
+
+			buildConfigRuntime.Labels[LabelKeyBuildType] = string(buildType)
+			updateRuntime = true
+		}
+
+		runtimeImageName, runtimeImageNamespace := parseImage(&runtimeImage)
+		if buildConfigRuntime.Spec.Strategy.SourceStrategy.From.Name != runtimeImageName {
+			log.Debug("Changes detected in 'ImageName'.", " OLD - ", buildConfigRuntime.Spec.Strategy.SourceStrategy.From.Name, " NEW - ", runtimeImageName)
+
+			buildConfigRuntime.Spec.Strategy.SourceStrategy.From.Name = runtimeImageName
+			updateRuntime = true
+		}
+		if buildConfigRuntime.Spec.Strategy.SourceStrategy.From.Namespace != runtimeImageNamespace {
+			log.Debug("Changes detected in 'ImageNamespace'.", " OLD - ", buildConfigRuntime.Spec.Strategy.SourceStrategy.From.Namespace, " NEW - ", runtimeImageNamespace)
+
+			buildConfigRuntime.Spec.Strategy.SourceStrategy.From.Namespace = runtimeImageNamespace
+			updateRuntime = true
+		}
+
+		if updateRuntime {
+			log.Info("Updating BuildConfig for Runtime")
+
+			if err := kubernetes.ResourceC(client).Update(buildConfigRuntime); err != nil {
+				return updateRuntime, err
+			}
+
+			if _, err := openshift.BuildConfigC(client).TriggerBuild(buildConfigRuntime, instance.Name); err != nil {
+				return updateRuntime, err
+			}
+
+			return updateRuntime, nil
+		}
+	}
+
+	return false, nil
+}
+
+func envToMap(env []v1alpha1.Env) map[string]string {
+	envMap := map[string]string{}
+
+	for _, e := range env {
+		envMap[e.Name] = e.Value
+	}
+
+	return envMap
+}
+
+func envVarToMap(env []corev1.EnvVar) map[string]string {
+	envMap := map[string]string{}
+
+	for _, e := range env {
+		envMap[e.Name] = e.Value
+	}
+
+	return envMap
+}
+
+func mapToEnv(env map[string]string) (envVars []corev1.EnvVar) {
+	for key, value := range env {
+		envVars = append(envVars, corev1.EnvVar{
+			Name:  key,
+			Value: value,
+		})
+	}
+
+	return envVars
```

Comment:
Could `envVarToMap` and `mapToEnv` functions be moved to `pkg/util/resources.go` file? I guess we have some functions there that are doing almost the same. Even an `env_var.go` file there would be useful.

Reply:
moved the functions to [env_var.go](https://github.com/kiegroup/kogito-cloud-operator/blob/33e9918d0604a0593a609031c21facc071cb286f/pkg/util/env_var.go#L23) and [envs.go](https://github.com/kiegroup/kogito-cloud-operator/blob/33e9918d0604a0593a609031c21facc071cb286f/pkg/util/envs.go#L51)